src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.AnalyzerExecutionContext.cs

```diff
@@ -21,44 +23,44 @@ private sealed class AnalyzerExecutionContext
             /// <summary>
             /// Map from (symbol, analyzer) to count of its member symbols whose symbol declared events are not yet processed.
             /// </summary>
-            private Dictionary<ISymbol, HashSet<ISymbol>> _lazyPendingMemberSymbolsMapOpt;
+            private Dictionary<ISymbol, HashSet<ISymbol>?>? _lazyPendingMemberSymbolsMap;
```

Comment:
💡 Consider refactoring the code to make _missing_ have the semantics that currently is assigned to `null`. The type would change to the following, with simpler usage:

```suggestion
            private Dictionary<ISymbol, HashSet<ISymbol>>? _lazyPendingMemberSymbolsMap;
```

Reply:
> Consider refactoring the code to make missing have the semantics that currently is assigned to null

I think that might be a risky change, as we track of whether or not all the member symbols have been added to this map to determine when to execute symbol end actions for container symbols. Even if some cleanup was possible here with small semantic changes, I'd like to avoid it in this PR, which currently avoids any semantic changes.